### PR TITLE
Update conversation participants outside of the transaction.

### PR DESCRIPTION
**Pull request by lasryaric**
## Description

When posting or editing a user message, the list of participants was being updated inside the `rank` locked transaction, which leads to holding the lock for longer than necessary. This PR moves the participants update code outside of the locked transaction.

### CODE REVIEW

File: front/lib/api/assistant/conversation.ts [hunk 1]
```diff
@@ -620,6 +620,29 @@ export async function* postUserMessage(
...
+      } else {
+        return ConversationParticipant.create({
+          conversationId: conversation.id,
+          userId: user.id,
+          action: "posted",
+        });
+      }
+    }
+  }
+  await createOrUpdateParticipation();
```
Comment (spolu) [inline front/lib/api/assistant/conversation.ts:645]: You add a sequential call to the db on the critical path here.

Can we parallelize with something else?

Comment (lasryaric) [reply front/lib/api/assistant/conversation.ts:645]: I think we are talking about less than 1ms or 2ms at worst. Do you think it's worst doing this optimization?

Comment (lasryaric) [reply front/lib/api/assistant/conversation.ts]: Adding a commit for //

**Review by spolu — APPROVED** [review]
Sorry for the delay!

**Pull request merged**